### PR TITLE
Fix --append option

### PR DESCRIPTION
**Pull request by jverdeyen**
Related to #306

### CODE REVIEW

**Comment by theofidry** [comment]
Thanks for the PR @jverdeyen :)

Did you try to run it manually? I think I'm missing an integration + unit test here you didn't have to change any tests and they are still green...

**Comment by jverdeyen** [comment]
Okay, I'll take a look at this. I'm currently using my patched version and it's working.

**Comment by theofidry** [comment]
Need to double check with https://github.com/theofidry/AliceDataFixtures/pull/32.

Sorry for the time taken @jverdeyen, I'll have time to check it soon.

**Comment by jverdeyen** [comment]
@theofidry no problem, I'm using my patched version in development for now. As fixtures are for development ;)

File: tests/Console/Command/Doctrine/LoadDataFixturesCommandIntegrationTest.php [hunk 1]
```diff
@@ -178,6 +199,21 @@ public function loadCommandProvider()
...
             [
+                '--append' => null,
+            ],
+            <<<'EOF'
+              > fixtures found:
+      - /home/travis/build/theofidry/AliceBundle/tests/Functional/TestBundle/Resources/fixtures/brand.yml
+      - /home/travis/build/theofidry/AliceBundle/tests/Functional/TestBundle/Resources/fixtures/product.yml
+      - /home/travis/build/theofidry/AliceBundle/tests/Functional/TestBundle/Bundle/ABundle/Resources/fixtures/aentity.php
+      - /home/travis/build/theofidry/AliceBundle/tests/Functional/TestBundle/Bundle/BBundle/Resources/fixtures/bentity.yml
+  > purging database
```
Comment (WillGibson) [inline tests/Console/Command/Doctrine/LoadDataFixturesCommandIntegrationTest.php]: It should not be saying "> purging database" when using --append. I tried taking this out locally and the test still passed, which surprised me. Hence comment I'm about to make further up re `$commandTester->getDisplay()`.

File: tests/Console/Command/Doctrine/LoadDataFixturesCommandIntegrationTest.php [hunk 2]
```diff
@@ -159,6 +159,27 @@ public function testFixturesRegisteringUsingMySQL(array $inputs, string $expecte
         $this->assertFixturesDisplayEquals($expected, $commandTester->getDisplay());
     }
 
+    /**
+     * @dataProvider loadCommandProvider
+     */
+    public function testFixturesUsingAppend(array $inputs, string $expected)
```
Comment (WillGibson) [inline tests/Console/Command/Doctrine/LoadDataFixturesCommandIntegrationTest.php]: This test is not needed, it's covered by `testFixturesRegisteringUsingMySQL`.

File: tests/Console/Command/Doctrine/LoadDataFixturesCommandIntegrationTest.php [hunk 3]
```diff
@@ -159,6 +159,27 @@ public function testFixturesRegisteringUsingMySQL(array $inputs, string $expecte
         $this->assertFixturesDisplayEquals($expected, $commandTester->getDisplay());
```
Comment (WillGibson) [inline tests/Console/Command/Doctrine/LoadDataFixturesCommandIntegrationTest.php]: @theofidry `$commandTester->getDisplay()` is returning an empty string and `$this->assertFixturesDisplayEquals` is passing regardless of whether the expected and actual match, which might be of some concern!

**Comment by WillGibson** [comment]
@theofidry I have manually tested the changes here in \Hautelook\AliceBundle\Loader\DoctrineOrmLoader.php with this command...

```
./bin/console hautelook:fixtures:load -n --append --env=dev
```
...and it works fine.

Looks like there's an issue with the second build job at https://travis-ci.org/hautelook/AliceBundle/builds/192958087. I'm not sure how to fix that, but all the tests are passing for me locally.

I've made some comments on https://github.com/hautelook/AliceBundle/pull/308/files

**Comment by theofidry** [comment]
Hm, to be checked with https://github.com/hautelook/AliceBundle/pull/329/files and https://github.com/theofidry/AliceDataFixtures/pull/32. All seem to relate to the same issue.

**Comment by nlzet** [comment]
@theofidry Hi, is there any update on this? It seems like currently loading fixtures with the --append flag is just broken. 

Running with truncate -> OK
Running with append -> "Cannot append loaded fixtures and at the same time purge the database. Choose one." (which happens because there is a null check instead of a boolean true check)

If you would only remove the "!== null" part it is already fixed.

**Comment by theofidry** [comment]
Closed by #345

**Pull request closed without merging**